packages/amplify-provider-awscloudformation/src/pre-push-cfn-processor/cfn-pre-processor.ts

```diff
@@ -24,3 +37,140 @@ export async function preProcessCFNTemplate(filePath: string): Promise<string> {
+  }
+
+  await writeCFNTemplate(cfnTemplate, filePath, { templateFormat });
+}
+
+
+//validate the format of actions and ARNs for custom IAM policies
+export function validateCustomPolicy(customPolicy: CustomIAMPolicy, resourceName: string) {
+  const resources = customPolicy.Resource;
+  const actions = customPolicy.Action;
+  let resourceRegex = new RegExp('arn:(aws[a-zA-Z0-9-]*):([a-zA-Z0-9\\-])+:([a-z]{2}(-gov)?-[a-z]+-\\d{1})?:(\\d{12})?:(.*)');
+  let actionRegex = new RegExp('[a-zA-Z0-9]+:[a-z|A-Z|0-9|*]+');
+  let wrongResourcesRegex = [];
+  let wrongActionsRegex = [];
+  let errorMessage = "";
+
+  for (const resource of resources) {
+    if (!resourceRegex.test(resource)) {
+      wrongResourcesRegex.push(resource);
+    }
+  }
+
+  for (const action of actions) {
+    if (!actionRegex.test(action)) {
+      wrongActionsRegex.push(action);
+    }
+  }
+
+  if (wrongResourcesRegex.length > 0) {
+    errorMessage += `\nInvalid ARN format for custom IAM policies in ${resourceName}:\n${wrongResourcesRegex.toString()}\n`;
+  }
+  if (wrongActionsRegex.length > 0) {
+    errorMessage += `\nInvalid actions format for custom IAM policies in ${resourceName}:\n${wrongActionsRegex.toString()}\n`;
+  }
+
+  if (errorMessage.length > 0) {
+    throw new CustomPoliciesFormatError(errorMessage);
+  }
+}
+
+
+//replace or add env parameter in the front of the resource customers enter to the current env
+export async function replaceEnvForCustomPolicies(policy: CustomIAMPolicy, currentEnv: string) : Promise<CustomIAMPolicy> {
+  let resourceWithEnv = [];
+  let action = policy.Action;
+  let effect = policy.Effect;
+  let customIAMpolicy: CustomIAMPolicy = {
+    Action: action,
+    Effect: effect,
+    Resource: []
+  }
+  for (let resource of policy.Resource){
+    resourceWithEnv.push(resource.replace( '{env}', currentEnv));
+  }
+  customIAMpolicy.Resource = resourceWithEnv;
+  return customIAMpolicy;
+}
+
+
+export async function validateCustomPoliciesSchema(data: CustomIAMPolicies, categoryName: string, resourceName: string) : Promise<Boolean> {
+  const ajv = new Ajv();
+  //validate if the policies match the custom IAM policies schema, if not, then not write into the CFN template
+  const validatePolicy = ajv.compile(CustomIAMPolicySchema);
+  if(!validatePolicy(data)) {
+    printer.warn(`The format of custom IAM policies in the ${categoryName} ${resourceName} is not valid`);
+    return false;
+  }
+  //the policies without env will be carried over and merge into the CFN template
+  return true;
+}
+
+export function warnWildCatCustomPolicies(customPolicies: CustomIAMPolicy[], resourceName: string,) {
+  for (const policy of customPolicies) {
+    const resources = policy.Resource;
+    for (const resource of resources) {
+      if(resource.includes('*')) {
+        printer.warn(`Warning: You've specified "*" as a custom IAM policy for your ${resourceName}.\n This will give your ${resourceName} access to ALL resources in the AWS Account.`)
```

Comment:
```suggestion
        printer.warn(`Warning: You've specified "*" as the resource in a custom IAM policy for ${resourceName}.\n This will give ${resourceName} access to ALL resources in the AWS Account.`)
```